### PR TITLE
feat(dataset): support annotations and data type for dataset

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- related: https://github.com/star-whale/starwhale/issues/960
- provide annotations and data type
- refactor ppl and cmp
- update example with new API

## Feature Show
- model eval
image
image

## Modules
- [x] Client
- [x] Python-SDK

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/model.py [hunk 1]
```diff
@@ -316,86 +257,47 @@ def _starwhale_internal_run_ppl(self) -> None:
...
-            except Exception as e:
-                self._sw_logger.exception(f"{label.data!r} label handle exception:{e}")
-                if not self.ignore_error:
-                    self._update_status(STATUS.FAILED)
-                    raise
-                else:
-                    _label = ""
-        # self._sw_logger.debug(f"record ppl result:{data.idx}")
+        annotations = {} if self.ignore_annotations else annotations
+        _b64: t.Callable[[bytes], str] = lambda x: base64.b64encode(x).decode("ascii")
```
Comment (goldenxinxing) [inline client/starwhale/api/_impl/model.py:296]: is it better to move this to class or global scope?

Comment (tianweidut) [reply client/starwhale/api/_impl/model.py:296]: base64 encode is already a function. We use a lambda function only as a shortcut.

Comment (goldenxinxing) [reply client/starwhale/api/_impl/model.py:296]: ok

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**